pegs_common_utils.R

```diff
@@ -262,63 +262,374 @@ prepare_ukb_phenotype <- function(ukb_data, phenotype) {
 # Returns:-
 # A dataframe (epr_number = participant ID, Y = phenotype). All participants to
 # be excluded have their phenotype marked as NA.
-prepare_pegs_phenotype <- function(pegs_data, phenotype) {
+prepare_pegs_phenotype <- function(pegs_data, 
+                                   phenotype = c("lower_gi_polyps",'fibroids',
+                                                 'boneloss','migraines','IDA',
+                                                 'ovariancysts','asthma',"T2D","Allergic_Rhinitis",
+                                                 'Stroke','Heart_Attack','Arrhythmia','CAD','CHF',
+                                                 'Cholesterol','Hypertension','AVSD','Angina','Angioplasty',
+                                                 'PBF','Bclots','Atherogenic')) {
   pheno_data <- NULL
 
-  # Type 2 diabetes (T2D)
-  if (phenotype == "T2D") {
-    # Exclude participants with likely different disease etiology, i.e. those
-    # with age at onset < 20 (these are likely Type 1 diabetes or MODY ppts) or
-    # those with gestational diabetes only
-    pegs_data <- pegs_data %>%
-      mutate(onset_age_under_20 = (if_else(he_c022e_diabetes_age_CHILDQ < 20, 1, 0)))
-
-    diabetes_exclusions <- pegs_data %>%
-      filter(he_c022_diabetes_PARQ == 1) %>%
-      filter(onset_age_under_20 == 1 |
-        is.na(onset_age_under_20)) %>%
-      pull(epr_number)
-    diabetes_exclusions <- sort(unique(c(
-      diabetes_exclusions,
-      (pegs_data %>%
-        filter(he_c022_diabetes_PARQ == 1 &
-          .data[["_he_gender_"]] %in% c(1, NA)) %>%
-        filter(he_c022a_diabetes_preg_CHILDQ == 1) %>%
-        pull(epr_number))
-    )))
-
-    pegs_data <- pegs_data %>%
-      mutate(Y = replace(
-        he_c022_diabetes_PARQ,
-        epr_number %in% diabetes_exclusions,
-        NA
-      ))
-
-    # Put all variables that were used for phenotype preparation into a
-    # dataframe
-    pheno_data <- pegs_data %>%
-      select(all_of(c(
-        "epr_number", "he_c022_diabetes_PARQ", "onset_age_under_20",
-        "_he_gender_", "he_c022a_diabetes_preg_CHILDQ", "Y"
-      )))
-  } else {
-    stop(paste("Unrecognized phenotype string :"), phenotype)
+  .lower_gi_phenotype <- function(pegs_data){
+    
+    cancer_cols <- names(pegs_data)[grep('cancer',names(pegs_data),ignore.case = T)]
+    cancer_ex <- c('age|sis|bro|dad|mom')
+    cancer_cols <- cancer_cols[-grep(cancer_ex,cancer_cols,ignore.case = T)]
+    gi_cols <- c('he_f038_lactose_intolerance','he_f039_crohns','he_f040_ulcerative_colitis')
+    exclusion_cols <- c(cancer_cols,gi_cols)
+    ex_df <- pegs_data[c(exclusion_cols)] 
+    ex_df[] <- lapply(ex_df, function(x) as.numeric(as.character(x))) 
+    d <- ex_df %>% 
+      as.data.frame(.) %>% 
+      mutate(sum = rowSums(.,na.rm = T)) %>% dplyr::select(sum,everything())
+    epr_number <- pegs_data$epr_number
+    ex_df <- cbind(epr_number,d) %>% dplyr::select(epr_number,sum)
```

Comment:
these lines of code can be combined to:
mutate(...) %>% select(sum) %>% cbind(epr_number)
Applies to all functions.